### ISSUE TITLE
Please add default visibility option for directories

### ISSUE THREAD

**Issue by stupidkitty** [thephpleague/flysystem-bundle#68]
Local adapter creates only private disrectories :(
https://github.com/thephpleague/flysystem-bundle/blob/b00803da1e2e7cd8c146266e58213407c5183fdf/src/Adapter/Builder/LocalAdapterDefinitionBuilder.php#L73
At this line, you can add a second argument for directories visibility rules.
Link on factory method:
https://github.com/thephpleague/flysystem/blob/7cbbb7222e8d8a34e71273d243dfcf383ed6779f/src/UnixVisibility/PortableVisibilityConverter.php#L99

Consider this improvement suggestion

**Comment by mynameisbogdan** [comment]
Similar for FTP/SFTP adapters. 

`directoryPerm`, `permPublic` and `permPrivate` doesn't change anything in 2.x.

**Comment by flohw** [comment]
Hello there,

After my investigation on the issue I just closed, I come to the same conclusion that there is a missing argument on the builder.

@tgalopin @OskarStark As I saw you on latest commits, I hope this notification can reach you.

Let me know if I can provide a PR but I will need some clues as the `$options` passed to the builder only has the options for the adapter. Can we add a third argument? Is there any other solution I can't think of?

Thank you
